app/models/mixins/retirement_mixin.rb

```diff
@@ -201,14 +202,18 @@ def retired_event_name
   end
 
   def raise_retirement_event(event_name, requester = nil)
```

Comment:
@mkanoor @d-m-u Since raise_retirement_event is only called in 2 places, and we know in what context for each, can we setup and pass the requester into this method? Then, we would only have to setup the q_options from the requester and a lot of this code would go away.